tests/unit/unit1981.c

```diff
@@ -0,0 +1,104 @@
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) Daniel Stenberg, <daniel@haxx.se>, et al.
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ * SPDX-License-Identifier: curl
+ *
+ ***************************************************************************/
+#include "curlcheck.h"
+#include "http_aws_sigv4.h"
+#include "dynbuf.h"
+
+
+static CURLcode unit_setup(void)
+{
+  return CURLE_OK;
+}
+
+static void unit_stop(void)
+{
+}
+
+
+struct testcase {
+  const char *testname;
+  const char *query_part;
+  const char *canonical_query;
+};
+
+static const struct testcase testcases[] = {
+      {
+          "no-value",
+          "Param1=",
+          "Param1="
+      },
+      {
+        "blank-query-params",
+        "hello=a&b&c=&d",
+        "b=&c=&d=&hello=a"
+      },
+      {
+          "get-vanilla-utf8-query",
+          "\u1234=bar",
+          "%E1%88%B4=bar"
+      },
+      {
+          "get-vanilla-query-order-key-case",
+          "Param2=value2&Param1=value1",
+          "Param1=value1&Param2=value2"
+      },
+      {
+          "get-vanilla-query-unreserved",
+          "-._~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz=-._~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz",
+          "-._~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz=-._~0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+      },
+      {
+          "get-vanilla-empty-query-key",
+          "Param1=value1",
+          "Param1=value1"
+      },
+      {
+          "get-vanilla-query-order-encoded",
+          "Param-3=Value3&Param=Value2&%E1%88%B4=Value1",
+          "%E1%88%B4=Value1&Param=Value2&Param-3=Value3"
+      },
+};
+
+UNITTEST_START
+{
+ struct dynbuf canonical_query;
+
+ char buffer[1024];
+   size_t testcases_size = sizeof(testcases) / sizeof(testcases[0]);
+   size_t i;
+   int result;
+  for(i = 0;  i < testcases_size; i++) {
+   Curl_dyn_init(&canonical_query, CURL_MAX_HTTP_HEADER);
+
+   result = Curl_canon_query(testcases[i].query_part, &canonical_query);
+   result = snprintf(buffer, sizeof(buffer), "%s: Received \"%s\" and should be \"%s\"", testcases[i].testname, Curl_dyn_ptr(&canonical_query), testcases[i].canonical_query);
+   fail_unless(result >= 0 , "snprintf fails");
+   fail_unless(!strcmp(Curl_dyn_ptr(&canonical_query), testcases[i].canonical_query),
+   buffer);
+ 
+ }
+
+}
```

Comment:
```suggestion
UNITTEST_START
{
  struct dynbuf canonical_query;
  char buffer[1024];
  size_t i;
  int result;

  for(i = 0; i < CURL_ARRAYSIZE(testcases); i++) {
    Curl_dyn_init(&canonical_query, CURL_MAX_HTTP_HEADER);    
                                                              
    result = Curl_canon_query(testcases[i].query_part, &canonical_query);
    result = snprintf(buffer, sizeof(buffer), "%s: Received '%s'"
                      " and should be '%s'",
                      testcases[i].testname,
                      Curl_dyn_ptr(&canonical_query),
                      testcases[i].canonical_query);
    fail_unless(result >= 0, "snprintf fails");
    fail_unless(!strcmp(Curl_dyn_ptr(&canonical_query),
                        testcases[i].canonical_query),
                buffer);
  }
}
```
long lines, indent, use `CURL_ARRAYSIZE()`.

Applies to rest of newly added tests too.